Common/Data/UniverseSelection/FineFundamentalFilteredUniverse.cs

```diff
@@ -39,5 +40,20 @@ public FineFundamentalFilteredUniverse(Universe universe, Func<IEnumerable<FineF
         {
             FineFundamentalUniverse = new FineFundamentalUniverse(universe.UniverseSettings, universe.SecurityInitializer, fineSelector);
         }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FineFundamentalFilteredUniverse"/> class
+        /// </summary>
+        /// <param name="universe">The universe to be filtered</param>
+        /// <param name="fineSelector">The fine selection function</param>
+        public FineFundamentalFilteredUniverse(Universe universe, PyObject fineSelector)
+            : base(universe, universe.SelectSymbols)
+        {
+            Func<IEnumerable<FineFundamental>, Symbol[]> func;
+            if (fineSelector.TryConvertToDelegate(out func))
+            {
+                FineFundamentalUniverse = new FineFundamentalUniverse(universe.UniverseSettings, universe.SecurityInitializer, func);
+            }
```

Comment:
I think we need error conditions everytime we fail to convert a `PyObject` to a delegate type (func/action/ect)

Reply:
Perhaps we need a `ConvertToDelegate` helper method which throws an exception when `TryConvertToDelegate` returns false.